sw-calculator/src/main/java/org/egov/swcalculation/validator/SWCalculationWorkflowValidator.java

```diff
@@ -0,0 +1,102 @@
+import java.util.Map;
+import java.util.TimeZone;
+
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.swcalculation.util.CalculatorUtils;
+import org.egov.swcalculation.web.models.Property;
+import org.egov.swcalculation.web.models.SewerageConnection;
+import org.egov.swcalculation.web.models.workflow.ProcessInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class SWCalculationWorkflowValidator {
+
+    @Autowired
+    private CalculatorUtils util;
+
+    public Boolean nonMeterconnectionValidation(RequestInfo requestInfo, String tenantId, String connectionNo){
+        Map<String,String> errorMap = new HashMap<>();
+        Boolean genratedemand = true;
+        applicationValidation(requestInfo,tenantId,connectionNo,errorMap);
+        if(!CollectionUtils.isEmpty(errorMap)){
+            log.error("DemandGeneartionError", "Demand cannot be generated as sewerage connection with connection number "+connectionNo+" or property associated with it, is in workflow and not approved yet");
+            genratedemand=false;
+        }
+        return genratedemand;
+    }
+
+    public Map<String,String> applicationValidation(RequestInfo requestInfo,String tenantId,String connectionNo, Map<String,String> errorMap){
+        List<SewerageConnection> sewerageConnectionList = util.getSewerageConnection(requestInfo,connectionNo,tenantId);
+        int size = sewerageConnectionList.size();
+        SewerageConnection sewerageConnection = sewerageConnectionList.get(size-1);
+        String sewerageApplicationNumber = sewerageConnection.getApplicationNo();
+        //Long dateEffectiveFrom = sewerageConnection.getDateEffectiveFrom();
+        sewerageConnectionValidation(requestInfo,tenantId,sewerageApplicationNumber,errorMap);
+        String propertyId = sewerageConnection.getPropertyId();
+        Property property = util.getProperty(requestInfo,tenantId,propertyId);
+        String propertyApplicationNumber = property.getAcknowldgementNumber();
+        propertyValidation(requestInfo,tenantId,propertyApplicationNumber,errorMap);
+        
+        /*if(!StringUtils.isEmpty(dateEffectiveFrom))
+            dateValidation(dateEffectiveFrom,connectionNo,errorMap);*/
+        return  errorMap;
+    }
+
+    public void sewerageConnectionValidation(RequestInfo requestInfo,String tenantId, String sewerageApplicationNumber,Map<String,String> errorMap){
+        Boolean isApplicationApproved = workflowValidation(requestInfo,tenantId,sewerageApplicationNumber);
+        if(!isApplicationApproved)
+            errorMap.put("sewerageApplicationError","Demand cannot be generated as sewerage connection application with application number "+sewerageApplicationNumber+" is in workflow and not approved yet");

```

Comment:
Error code should be in caps seperated by underscore. Please update at all the places

Reply:
will update this